### PR TITLE
move inline js for author autocomplete

### PR DESCRIPTION
**Pull request by RayBB**
Move autocomplete part of inline js for `openlibrary/templates/books/author-autocomplete.html` #4474

refactor

### Technical

Nothing interesting in this PR.
Following the standards set for previous tickets in epic.

### Testing

* Got to a [book edit page](https://openlibrary.org/books/OL7455361M/Biological_Science/edit) and try adding/removing authors 

### Screenshot

https://user-images.githubusercontent.com/921217/118733787-adb3e900-b7d8-11eb-9873-ce111a60f205.mp4

### Stakeholders

@jdlrobson

### CODE REVIEW

File: openlibrary/templates/books/author-autocomplete.html [hunk 1]
```diff
@@ -38,28 +38,8 @@
         <a href="javascript:;" class="remove red plain" title="$_('Remove this author')">[&times;]</a>&nbsp;
         <a href="javascript:;" class="add">$_("Add another author?")</a>
     </div>
-
-<div id="authors">
```
Comment (jdlrobson) [inline openlibrary/templates/books/author-autocomplete.html:42]: MINOR UI REGRESSION:

Removing this ID  causes the cross icon to jump to a new line for me. Let's make sure the rule `#authors div, div.lists` still applies - potentially by adding `multi-input-autocomplete--author` to static/css/page-user.less for now.


Screen Shot 2021-05-23 at 10 25 35 AM

File: openlibrary/plugins/openlibrary/js/index.js [hunk 2]
```diff
@@ -93,25 +104,46 @@ jQuery(function () {
...
+    const excerpts = document.getElementById('excerpts');
+    const links = document.getElementById('links');
+
+    if (
+        addRowButton ||
+        autocompleteLanguage || autocompleteWorks || excerpts || links
+    ) {
         import(/* webpackChunkName: "user-website" */ './edit')
-            .then(module => module.initWorksMultiInputAutocomplete());
+            .then(module => {
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:119]: FYI: I merged into a single entry point here.

File: openlibrary/plugins/openlibrary/js/edit.js [hunk 3]
```diff
@@ -72,6 +75,29 @@ export function initWorksMultiInputAutocomplete() {
...
+    getJqueryElements('.multi-input-autocomplete--author').forEach(jqueryElement => {
+        /* Values in the html passed from Python code */
+        const dataConfig = JSON.parse(jqueryElement[0].dataset.config);
+        jqueryElement.setup_multi_input_autocomplete(
+            'input.author-autocomplete',
+            render_author.bind(null, dataConfig.name_path, dataConfig.dict_path, false),
+            {
+                endpoint: '/authors/_autocomplete',
+                // Don't render "Create new author" if searching by key
+                addnew: function(query) { return !/^OL\d+A/i.test(query); },
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/edit.js:89]: Consider using shorthand e.g. `(query)=> !/^OL\d+A/i.test(query)`

File: openlibrary/plugins/openlibrary/js/index.js [hunk 4]
```diff
@@ -1,3 +1,14 @@
+export function initAddBookImport () {
+    $('.list-books a').click(function() {
+        var li = $(this).parents('li:first');
+        $('input#work').val(`/works/${li.attr('id')}`);
+        $('form#addbook').submit();
+    });
+    $('#bookAddCont').click(function() {
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:7]: FYI (for future patches): https://github.com/internetarchive/openlibrary/issues/5198

**Review by jdlrobson — DISMISSED** [review]
UI regression is the only remaining blocker to merging this one.

File: openlibrary/plugins/openlibrary/js/index.js (hunk 4 above)
Comment (RayBB) [reply openlibrary/plugins/openlibrary/js/index.js]: I think you accidentally added that function. I removed it

**Comment by RayBB** [comment]
@jdlrobson good catch on the visual regression. Went ahead and changed the .less file to use the class and it works great now. Also addressed you other comments.

File: openlibrary/plugins/openlibrary/js/index.js (hunk 4 above)
Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/index.js]: Thanks!

**Pull request merged**